### PR TITLE
Reset ACSSeqNum stored in state file when containerInstance …

### PR DESCRIPTION
**Pull request by kiranmeduri**
…is reset

This fix will prevent from agents connecting using bad ACSSeqNum when agent is registered as new Container Instance.
This usually happens when EC2 instance (registered as Container Instance) is snapshotted and launched as a new EC2 instance i
(registered as new Container Instance). Before this fix the ACSSeqNum of the previous Container Instance is used for the new one.

### CODE REVIEW

File: agent/agent.go [hunk 1]
```diff
@@ -264,7 +264,8 @@ func initializeStateManager(cfg *config.Config, taskEngine engine.TaskEngine, cl
 		statemanager.AddSaveable("ContainerInstanceArn", containerInstanceArn),
 		statemanager.AddSaveable("Cluster", cluster),
 		statemanager.AddSaveable("EC2InstanceID", savedInstanceID),
-		statemanager.AddSaveable("ACSSeqNum", sequenceNumber),
+		//not tracking ACSSeqNum for now
+		statemanager.AddSaveable("ACSSeqNum", utilatomic.NewIncreasingInt64(1)),
```
Comment (aaithal) [inline agent/agent.go]: Can this just be an int?

File: agent/acs/handler/acs_handler.go [hunk 2]
```diff
@@ -185,7 +180,7 @@ func acsWsURL(endpoint, cluster, containerInstanceArn string, taskEngine engine.
 	query.Set("containerInstanceArn", containerInstanceArn)
 	query.Set("agentHash", version.GitHashString())
 	query.Set("agentVersion", version.Version)
-	query.Set("seqNum", strconv.FormatInt(SequenceNumber.Get(), 10))
+	query.Set("seqNum", strconv.FormatInt(1, 10))
```
Comment (aaithal) [inline agent/acs/handler/acs_handler.go]: Can you add a unit test or extend the existing one for this? https://github.com/aws/amazon-ecs-agent/blob/dev/agent/acs/handler/acs_handler_test.go#L56

**Comment by aaithal** [comment]
lgtm overall. I have a couple of minor comments

**Comment by aaithal** [comment]
:shipit:

File: agent/agent.go [hunk 3]
```diff
@@ -264,7 +263,8 @@ func initializeStateManager(cfg *config.Config, taskEngine engine.TaskEngine, cl
 		statemanager.AddSaveable("ContainerInstanceArn", containerInstanceArn),
 		statemanager.AddSaveable("Cluster", cluster),
 		statemanager.AddSaveable("EC2InstanceID", savedInstanceID),
-		statemanager.AddSaveable("ACSSeqNum", sequenceNumber),
+		//not tracking ACSSeqNum for now
```
Comment (samuelkarp) [inline agent/agent.go]: What does "not tracking" mean here?  If `ACSSeqNum` is only preserved to avoid changing the state file version, can you rephrase to indicate that?

Comment (kiranmeduri) [inline agent/agent.go]: Any suggestions? I can change it to 

*ACSSeqNum is always pinned to '1'. In future we may want to deprecate this field completely from agent's state.*

Comment (samuelkarp) [inline agent/agent.go]: How about: "The `ACSSeqNum` field is retained for compatibility with `statemanager.EcsDataVersion` 4 and can be removed in the future with a version bump."  Alternately: just remove it now and bump the version?

Comment (kiranmeduri) [inline agent/agent.go]: Updated the comment, will remove the field and bump the version after in future release (after I understand the implications of it). Thanks

File: agent/acs/handler/acs_handler.go (hunk 2 above)
Comment (samuelkarp) [inline agent/acs/handler/acs_handler.go]: You can get rid of the `strconv.FormatInt` and just replace it with`"1"`.

**Comment by samuelkarp** [comment]
:+1:
